aip/0126.md

```diff
@@ -0,0 +1,102 @@
+---
+aip:
+  id: 126
+  state: reviewing
+  created: 2019-07-24
+permalink: /126
+redirect_from:
+  - /0126
+---
+
+# Enumerations
+
+It is common for a field to only accept or provide a discrete and limited set
+of values. In these cases, it can be useful to use enumerations (generally
+abbreviated "enums") in order to clearly communicate what the set of allowed
+values are.
+
+## Guidance
+
+APIs **may** expose enum objects for sets of values that are expected to change
+infrequently:
+
+```proto
+// A representation of a book.
+message Book {
+  // Other fields...
+
+  // Possible editions that may be published.
+  enum Edition {
+    // Default value. This value is unused.
+    EDITION_UNSPECIFIED = 0;
+
+    // The printed edition, in hardback.
+    HARDBACK = 1;
+
+    // The printed edition, in paperback.
+    PAPERBACK = 2;
+
+    // An e-book edition.
+    EBOOK = 3;
+
+    // An audio recording.
+    AUDIOBOOK = 4;
+  }
+
+  // The edition of the book.
+  Edition edition = 99;
+
+  // Other fields...
+}
+```
+
+- All enum values **must** use `SCREAMING_SNAKE_CASE`.
+- The first value of the enum **should** be the name of the enum itself
```

Comment:
Should we have something here about uniqueness requirements? In particular the common convention of using the enum name as a prefix for *all* values in an enum to provide uniqueness when there are multiple enums which would otherwise have the same value names.

Reply:
Done.